packages/store/src/lib/StoreSchema.ts

```diff
@@ -62,16 +108,30 @@ export class StoreSchema<R extends UnknownRecord, P = unknown> {
 		return new StoreSchema<R, P>(types as any, options ?? {})
 	}
 
+	readonly migrations: Record<string, MigrationSequence> = {}
+	readonly sortedMigrations: readonly Migration[]
+
 	private constructor(
 		public readonly types: {
 			[Record in R as Record['typeName']]: RecordType<R, any>
 		},
 		private readonly options: StoreSchemaOptions<R, P>
-	) {}
+	) {
+		for (const m of options.migrations ?? []) {
+			assert(!this.migrations[m.sequenceId], `Duplicate migration sequenceId ${m.sequenceId}`)
+			validateMigrations(m)
+			this.migrations[m.sequenceId] = m
+		}
+		const allMigrations = Object.values(this.migrations).flatMap((m) => m.sequence)
+		this.sortedMigrations = sortMigrations(allMigrations)
 
-	// eslint-disable-next-line no-restricted-syntax
-	get currentStoreVersion(): number {
-		return this.options.snapshotMigrations?.currentVersion ?? 0
+		for (const migration of this.sortedMigrations) {
+			if (!migration.dependsOn?.length) continue
+			for (const dep of migration.dependsOn) {
+				const depMigration = allMigrations.find((m) => m.id === dep)
+				assert(depMigration, `Migration '${migration.id}' depends on missing migration '${dep}'`)
```

Comment:
I later ran into an issue here with bindings. If I have a store migration that depends on a particular arrow migration but I'm using the editor without the arrows shape it blows up 😬

not sure what the right move here is. i think ideally i'd write the store migration that splits out bindings within the sequence for arrows themselves? not sure that's an option though...

Reply:
it's an option. I actually had an earlier commit where I made it so the `shapeUtil.migrations` property could be a standard `MigrationSequence` as an alternative to the special props migration sequence. I dropped it to avoid polluting the API but sounds like we already have a good use case for it.